org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.java

```diff
@@ -173,14 +173,6 @@ public void initialize(List<WorkspaceFolder> workspaceFolders, Procedure2<? supe
 		refreshWorkspaceConfig(cancelIndicator);
 	}
 
-	/**
-	 * @return whether this workspace manager supports multiple workspace root folders.
-	 * @since 2.21
-	 */
-	public boolean isSupportsWorkspaceFolders() {
```

Comment:
wonder if we should keep this and also in the check, @szarnekow what do you think? do we want to opt out from workspace folders on server side=

Reply:
I've seen language server implementations that explicitly wanted to disable these, yes.

Reply:
How would you opt out from this? By simply overriding this method?

Reply:
Or by some sort of configuration?

Reply:
I think the negotiation between client and server is done by means of the InitializeResult. If the server doesn't support folders, it would set the boolean value to false (not to true, respectively).